azext_edge/edge/providers/orchestration/resources/connector/opcua/certs.py

```
@@ -0,0 +1,494 @@
+        cl_resources = self._get_cl_resources(instance_name=instance_name, resource_group=resource_group)
+        secretsync_spc = self._find_existing_spc(instance_name=instance_name, cl_resources=cl_resources)
+        if not secretsync_spc:
+            return
+
+        # get properties from default spc
+        spc_properties = secretsync_spc.get("properties", {})
+        spc_keyvault_name = spc_properties.get("keyvaultName", "")
+        spc_tenant_id = spc_properties.get("tenantId", "")
+        spc_client_id = spc_properties.get("clientId", "")
+
+        self.keyvault_client = get_keyvault_client(
+            subscription_id=self.default_subscription_id,
+            keyvault_name=spc_keyvault_name,
+        )
+
+        secrets: PageIterator = self.keyvault_client.list_properties_of_secrets()
+
+        # get file extension
+        file_name = os.path.basename(file)
+        cert_extension = validate_file_extension(file_name, [".der", ".crt", ".crl"])
+        # get cert name by removing extension
+        cert_name = file_name.replace(f"{cert_extension}", "")
+
+        try:
+            opcua_secret_sync = self.ssc_mgmt_client.secret_syncs.get(
+                resource_group_name=resource_group,
+                secret_sync_name=OPCUA_ISSUER_LIST_SECRET_SYNC_NAME,
+            )
+        except ResourceNotFoundError:
+            opcua_secret_sync = {}
+
+        if cert_extension == ".crl":
+            if opcua_secret_sync:
+                secret_mapping = opcua_secret_sync.get("properties", {}).get("objectSecretMapping", [])
+                possible_file_names = [f"{cert_name}.crt", f"{cert_name}.der"]
+                matched_names = [
+                    mapping["targetKey"] for mapping in secret_mapping if mapping["targetKey"] in possible_file_names
+                ]
+
+            if not opcua_secret_sync or not matched_names:
+                logger.error(f"Cannot add .crl {file_name} without corresponding .crt or .der file.")
```

you can just raise the error...

---

This would be some form of ArgumentError - I can't remember the exact enums but probably some kind of "required argument missing" error